packages/agent/src/agent/http/index.ts

```diff
@@ -288,7 +298,11 @@ export class HttpAgent implements Agent {
     this.#fetch = options.fetch || getDefaultFetch() || fetch.bind(global);
     this.#fetchOptions = options.fetchOptions;
     this.#callOptions = options.callOptions;
-    this.rootKey = options.rootKey ? options.rootKey : fromHex(IC_ROOT_KEY);
+
+    this.#rootKey = options.rootKey ?? fromHex(IC_ROOT_KEY);
+    this.#shouldFetchRootKey = options.shouldFetchRootKey ?? false;
+    // kick off the fetchRootKey process asynchronously, if needed
+    (async () => await this.fetchRootKey())();
```

Comment:
We start fetching the root key as soon as possible, in practice this should mean that there are no further delays compared to creating the agent with an async constructor and waiting for this task to complete before making API calls.

Reply:
Agreed that functionally the two approaches are the same.

Doing it this way was a stylistic choice to make it very explicit to any reader of the code that we're invoking a promise but not awaiting it on purpose. I'm fine with just invoking the promise.